src/augeas.c

```diff
@@ -2020,6 +2021,35 @@ int aug_escape_name(augeas *aug, const char *in, char **out) {
     return result;
 }
 
+int aug_load_file(struct augeas *aug, const char *file) {
+    int result = -1;
+    struct tree *meta = tree_child_cr(aug->origin, s_augeas);
+    struct tree *load = tree_child_cr(meta, s_load);
+    bool found = false;
+
+    api_entry(aug);
+
+    ERR_NOMEM(load == NULL, aug);
+
+    list_for_each(xfm, load->children)  {
+      if (filter_matches(xfm, file)) {
+        transform_load(aug, xfm, file);
```

Comment:
Each tree node has a `dirty` flag; whenever a tree node is modified, its `dirty` flag is set, as well as the `dirt` flags of all its ancestors. That is done by `tree_mark_dirty` in `augeas.c`. `tree_clean` does the reverse: clearing the flag on a node and all its descendants. `aug_save` (and `tree_save`, its internal cousin) only tries to save files that are marked as dirty to avoid lots of unnecessary tree -> file transformations.